cmd/admin-handlers.go

```diff
@@ -1031,25 +1031,43 @@ func toAdminAPIErr(ctx context.Context, err error) APIError {
 // - input entry is not of the type *trace.Info*
 // - errOnly entries are to be traced, not status code 2xx, 3xx.
 // - all entries to be traced, if not trace only S3 API requests.
-func mustTrace(entry interface{}, trcAll, errOnly bool) bool {
+func mustTrace(entry interface{}, trcS3, trcInternal, trcStorage, errOnly bool, threshold time.Duration) (shouldTrace bool) {
```

Comment:
```suggestion
func mustTrace(entry interface{}, trcS3, trcInternal, trcStorage, errOnly bool, threshold time.Duration) (shouldTrace bool) {
```

Take a data structure now such as TraceOptions{} @vadmeste